### PR TITLE
Divyanshu | Test | Prod sanity - Subscription Design points

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cvcu8bw

### CODE REVIEW

File: apps/web-giddh/src/app/shared/header/header.component.html [hunk 1]
```diff
@@ -2,7 +2,12 @@
 <div class="message-wrapper" *ngIf="showDepreciationMessage">
     <p class="text-center py-2">
         {{ localeData?.depreciation_message }}
-        <a href="mailto:support@giddh.com" aria-label="email" class="text-underline">support@giddh.com</a>
+        <a href="mailto:support@giddh.com" aria-label="email" class="text-underline">
+            support
+            <span>&#64;</span> <!-- This &#64; Html Code for @ symbol  -->
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/header.component.html:7]: remove commented code

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts [hunk 2]
```diff
@@ -281,8 +290,8 @@ export class BuyPlanComponent implements OnInit, OnDestroy {
                 this.commonCountrySource = [];
                 Object.keys(response).forEach(key => {
                     this.commonCountrySource.push({
-                        value: response[key].alpha2CountryCode,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.ts]: revert this code

File: apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.ts [hunk 3]
```diff
@@ -150,15 +150,6 @@ export class ViewSubscriptionComponent implements OnInit, OnDestroy {
         }
     }
 
-    /**
-     * Navigates back to the previous page.
-     *
-     * @memberof ViewSubscriptionComponent
-     */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.ts]: revert this code

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html [hunk 4]
```diff
@@ -28,27 +22,32 @@
...
+                                        <div class="country-dropdown d-flex flex-column" [ngClass]="{ 'mx-auto': !inputData?.length }">
+                                           
+                                            <img [src]="getFlagUrl(currentCountry.value?.additional?.alpha2CountryCode ? currentCountry.value?.additional?.alpha2CountryCode : currentCountry.value?.additional?.alpha3CountryCode)" [alt]="'IN -Inida'" width="32" height="32" class="flag cursor-pointer" (click)="countryList.open()">
+
+                                            <mat-select [name]="'country'" [formControl]="currentCountry" #countryList  (selectionChange)="newUserSelectCountry($event.value, false)"  panelClass="country-dropdown-list" [placeholder]="
+                                            commonLocaleData?.app_select + ' ' + commonLocaleData?.app_country">
+                                                <!-- <mat-select-trigger class="pr-2">
+                                                    {{ currentCountry.value?.additional?.countryName || newUserSelectedCountry || ''}}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html]: remove commented code

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html [hunk 5]
```diff
@@ -28,27 +22,32 @@
...
+                                                </mat-select-trigger> -->
+                                                <ng-container *ngFor="let country of countrySource">
+                                                    <mat-option [value]="country">
+                                                       <div class="d-flex w-100 align-items-center column-gap1">
+                                                            <img [src]="getFlagUrl(country.additional?.alpha2CountryCode ? country.additional?.alpha2CountryCode : country.additional?.alpha3CountryCode )" [alt]="country.label" width="24" height="16">
+                                                            <span>{{country.label}}</span>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html]:
```suggestion
                                                            <span>{{country?.label}}</span>
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/services/general.service.ts [hunk 6]
```diff
@@ -1953,4 +1953,15 @@ export class GeneralService {
...
     }
+
+    /**
+     * Get Country Flag Image Url by 2 digit country code
+     *
+     * @param {*} alpha2CountryCode
+     * @returns {string}
+     * @memberof GeneralService
+     */
+    public getCountryFlagUrl(alpha2CountryCode): string {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/services/general.service.ts]:
```suggestion
    public getCountryFlagUrl(countryCode: string): string {
```

File: apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html [hunk 7]
```diff
@@ -28,27 +22,29 @@
                             <div class="plan-width">
                                 <ng-template matStepLabel>{{ localeData?.all_plans }}</ng-template>
                                 <div class="d-flex justify-content-between align-items-center mb-3">
-                                    <h2 class="font-16 strong mr-3">{{ localeData?.choose_plans }}</h2>
-                                    <div class="d-flex column-gap15 align-items-center">
+                                    <div class="width-100 dropdown-wrapper" *ngIf="!(planListInProgress$ | async) && !isLoading && !isChangePlan">
+                                        <div class="country-dropdown d-flex flex-column" [ngClass]="{ 'mx-auto': !inputData?.length }">
+                                           
+                                            <img [src]="getFlagUrl(currentCountry.value?.additional?.alpha2CountryCode ? currentCountry.value?.additional?.alpha2CountryCode : currentCountry.value?.additional?.alpha3CountryCode)" [alt]="'IN -Inida'" width="32" height="32" class="flag cursor-pointer" (click)="countryList.open()">
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/subscription/buy-plan/buy-plan.component.html]: alt tag content should be dynamic

File: apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.html [hunk 8]
```diff
@@ -109,28 +109,55 @@ <h4 class="font-16 strong">
...
+                                    class="d-inline-block"
+                                    [amount]="viewSubscriptionData?.amount"
+                                    [currencySymbol]="viewSubscriptionData?.planCurrency?.symbol"
+                                    [currencyCode]="false"
+                                >
+                                </amount-field>
                             </span>
                         </div>
-                        <h4 class="strong border-bottom pb-3 font-16 mr-t2">Total</h4>
+                        <h4 class="border-bottom pb-3 font-16 mr-t2">Total</h4>
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/subscription/view-subscription/view-subscription.component.html]: translate total

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**